dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/persistence/ConsentAwareStorage.kt

```diff
@@ -53,20 +56,14 @@ internal class ConsentAwareStorage(
         forceNewBatch: Boolean,
         callback: (EventBatchWriter) -> Unit
     ) {
-        val orchestrator = when (datadogContext.trackingConsent) {
-            TrackingConsent.GRANTED -> grantedOrchestrator
-            TrackingConsent.PENDING -> pendingOrchestrator
-            TrackingConsent.NOT_GRANTED -> null
-        }
-
         val metric = internalLogger.startPerformanceMeasure(
             callerClass = ConsentAwareStorage::class.java.name,
             metric = TelemetryMetricType.MethodCalled,
             samplingRate = MethodCallSamplingRate.RARE.rate,
-            operationName = "writeCurrentBatch[${orchestrator?.getRootDirName()}]"
+            operationName = "writeCurrentBatch[$featureName]"
         )
-
         executorService.submitSafe("Data write", internalLogger) {
+            val orchestrator = resolveOrchestrator()
```

Comment:
we could do this yes, let me check what impact it will have if we call the `callback.invoke(writer)` outside of the `synchronised` block because to respect the flow we will still have to call that with a `NoOpWriter` in this case. I wonder if the minor cost we save here will not cause something else. WDYT ?